Userland/Libraries/LibArchive/Zip.h

```diff
@@ -271,9 +276,16 @@ class Zip {
 
 class ZipOutputStream {
 public:
-    ZipOutputStream(NonnullOwnPtr<Stream>);
+    using AddMemberCallback = Function<void(ZipMemberFromFileResult const&)>;
+
+    ZipOutputStream(NonnullOwnPtr<Stream> stream);
+
+    ErrorOr<void> add_member(ZipMember const& member);
+
+    ErrorOr<void> add_member_from_path(StringView path, bool recurse_through_directories, AddMemberCallback const& callback);
+    ErrorOr<void> add_directory_from_path(StringView path, bool recurse_through_directories, AddMemberCallback const& callback);
```

Comment:
Please make these boolean parameters a `Yes/No` enum class, which is much clearer than `add_member_from_path("/the/path", true, ...)`. Grep for `::Yes` to see some examples :)